addon/resolve-attribute-util.js

```diff
@@ -114,13 +116,32 @@ export function resolveValue(key, value, modelName, store, schema, record, paren
   const schemaInterface = recordData.schemaInterface;
 
   // First check to see if given value is either a reference or an array of references
-  let reference = computeAttributeReference(key, value, modelName, schemaInterface, schema);
-  if (reference !== undefined && reference !== null) {
+  let computedValue, nested, reference;
+  if (schema.useComputeAttribute()) {
```

Comment:
this is a hot path; at a minimum we will want to cache this

Reply:
:+1: cached in a weak map